pkg/gateway/multiparts/tracker.go

```diff
@@ -61,59 +101,30 @@ func (m *tracker) Get(ctx context.Context, uploadID string) (*MultipartUpload, e
 	if uploadID == "" {
 		return nil, ErrInvalidUploadID
 	}
-	res, err := m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		var m MultipartUpload
-		if err := tx.Get(&m, `
-			SELECT upload_id, path, creation_date, physical_address, metadata, content_type 
-			FROM gateway_multiparts
-			WHERE upload_id = $1`,
-			uploadID); err != nil {
-			return nil, err
-		}
-		return &m, nil
-	})
+	data := &MultipartUploadData{}
+	err := m.store.GetMsg(ctx, []byte(uploadID), data)
 	if err != nil {
 		return nil, err
 	}
-	return res.(*MultipartUpload), nil
+	return &MultipartUpload{
+		UploadID:        data.UploadId,
+		Path:            data.Path,
+		CreationDate:    data.CreationDate.AsTime(),
+		PhysicalAddress: data.PhysicalAddress,
+		Metadata:        data.Metadata,
+		ContentType:     data.ContentType,
+	}, nil
 }
 
 func (m *tracker) Delete(ctx context.Context, uploadID string) error {
 	if uploadID == "" {
 		return ErrInvalidUploadID
 	}
-	_, err := m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		res, err := tx.Exec(`DELETE FROM gateway_multiparts WHERE upload_id = $1`, uploadID)
-		if err != nil {
-			return nil, err
+	if _, err := m.Get(ctx, uploadID); err != nil {
```

Comment:
This is good for debug, but not for performance. We can just call Delete on the store.

Reply:
This is required since the current multiparts behavior relies on getting an error from delete in case the entry doesn't exist. I can change that but that will lead to behavior change

Reply:
Nice catch, I found it confusing in the original implementation but I guess this is how S3 behaves 🤷

Reply:
I assume we will have to verify if the S3 protocol does it too - if it is not the case we will change the behviour as there is no way to ensure that by calling Get a second before Delete.